lib/ansible/plugins/strategy/__init__.py

```diff
@@ -602,6 +634,8 @@ def search_handler_blocks_by_name(handler_name, handler_blocks):
                         self._add_group(original_host, result_item)
 
                     if 'ansible_facts' in result_item:
+                        # Set facts that should always be on the delegated hosts
+                        self._set_always_delegated_facts(result_item, original_task)
```

Comment:
Couldn't we also just completely skip this if we're delegating wholesale? (ie maybe it belongs in the `else` right below)

Reply:
Done.